lib/xray_creator.py

```diff
@@ -47,10 +47,7 @@ def _initialize_books(self):
         book_lookup = {}
         for book in self.books_not_failing():
             self._total_not_failing += 1
-            if book_lookup.has_key(book.title):
```

Comment:
Oh I was completely misunderstanding how defaultdict works. Lol.. Well it's
not needed anymore anyway but it's nice to know.. Thanks!

On Thu, Jun 23, 2016, 4:46 PM Anthony Toole <notifications@github.com>
wrote:

> In lib/xray_creator.py
> <https://github.com/szarroug3/X-Ray_Calibre_Plugin/pull/52#discussion_r68321191>
> :
>
> > @@ -47,10 +47,7 @@ def _initialize_books(self):
> >          book_lookup = {}
> >          for book in self.books_not_failing():
> >              self._total_not_failing += 1
> > -            if book_lookup.has_key(book.title):
>
> I think this should work. To be complete, I think the change is from this:
>
>         book_lookup = {}
>         for book in self.books_not_failing():
>             self._total_not_failing += 1
>             if book_lookup.has_key(book.title):
>                 book_lookup[book.title].append(book)
>                 continue
>             book_lookup[book.title] = [book]
>
> to this:
>
>         book_lookup = defaultdict{list}
>         for book in self.books_not_failing():
>             self._total_not_failing += 1
>             book_lookup[book.title].append(book)
>
> The fact self.books_not_failing is a genarator shouldn't matter - the loop
> handles one book at a time. Obviously if I'm wrong then say, always happy
> to learn :)
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/szarroug3/X-Ray_Calibre_Plugin/pull/52/files/09c125a697b713dc4d4ddc7ffa0dec29542c36bf#r68321191>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIEjUdrwUhDxFjRqvAv5ZXEzW1c3rXydks5qOv6mgaJpZM4I8-nd>
> .
>